warehouse/integrations/snowflake/snowflake.go

```diff
@@ -836,11 +842,14 @@ func (sf *Snowflake) LoadIdentityMappingsTable(ctx context.Context) error {
 	return nil
 }
 
-// ShouldAppend returns true if:
-// * the load table strategy is "append" mode
-// * the uploader says we can append
-func (sf *Snowflake) ShouldAppend() bool {
-	return sf.config.loadTableStrategy == loadTableStrategyAppendMode && sf.Uploader.CanAppend()
+// ShouldAppendTable returns true if:
+// * the load table strategy is "append" mode OR appendOnlyTables contains the table name
+// * AND the uploader says we can append
```

Comment:
```suggestion
// * the load table strategy is "append" mode OR appendOnlyTables contains the table name AND the uploader says we can append
// * OR the table name is in the appendOnlyTables configuration setting
```

Reply:
if we can not append, then appendOnlyTables is not checked.